### ISSUE TITLE
`prefer-function-over-method` should ignore abstract methods

### ISSUE THREAD

**Issue by sbj42** [palantir/tslint#2305]
`prefer-function-over-method` seems to warn "Class method does not use 'this'. Use a function instead." for abstract class methods.  Abstract class methods do not have implementations, and so of course they don't use 'this'.  `prefer-function-over-method` should ignore all abstract methods.

TSLint version: 4.4.2
TypeScript version: 2.2.1
Running TSLint via: CLI

The following code:
```ts
export abstract class PreferFunction {
    public abstract abstractMethod();
}
```

With the following tslint.json:
```json
{
    "extends": "tslint:latest",
    "rules": {
        "prefer-function-over-method": true
    }
}
```

Produces the following output:
```
example3.ts[2, 21]: Class method does not use 'this'. Use a function instead.
```

**Comment by ajafff** [comment]
You're right. It should ignore all methods without a body, including
overload signatures.

Am 06.03.2017 19:46 schrieb "James Clark" <notifications@github.com>:
